src/Microsoft.OpenApi.OData.Reader/Edm/RecordExpressionExtensions.cs

```diff
@@ -29,20 +29,10 @@ internal static class RecordExpressionExtensions
             Utils.CheckArgumentNull(record, nameof(record));
             Utils.CheckArgumentNull(propertyName, nameof(propertyName));
 
-            if (record.Properties != null)
-            {
-                IEdmPropertyConstructor property = record.Properties.FirstOrDefault(e => e.Name == propertyName);
-                if (property != null)
-                {
-                    IEdmIntegerConstantExpression value = property.Value as IEdmIntegerConstantExpression;
-                    if (value != null)
-                    {
-                        return value.Value;
-                    }
-                }
-            }
-
-            return null;
+            return (record.Properties?.FirstOrDefault(e => e.Name == propertyName) is IEdmPropertyConstructor property &&
```

Comment:
all of these code change are not related to the $count path feature. 
Maybe better to have a separate PR for clarification, better understanding.

Reply:
I'll try to keep that in mind for the next pull requests. I usually try to fix warnings/suggestions on the files I touch, but not the ones I don't touch. As this is my first actual PR in this codebase, I was exploring the code to understand how things work and I made a few modifications at other places. Since the work was already done, and it's only syntax change that doesn't impact the behavior, I suggest we make a one time exception. Thoughts?

Reply:
Oh, Ok